modules/402-ingress-nginx/images/kruise/patches/add-pdb.patch

```diff
@@ -249,3 +266,54 @@ index 1934f145..f5e7bbc1 100644
  func (dsc *ReconcileDaemonSet) storeDaemonSetStatus(ds *appsv1alpha1.DaemonSet, desiredNumberScheduled, currentNumberScheduled, numberMisscheduled, numberReady, updatedNumberScheduled, numberAvailable, numberUnavailable int, updateObservedGen bool, hash string) error {
  	if int(ds.Status.DesiredNumberScheduled) == desiredNumberScheduled &&
  		int(ds.Status.CurrentNumberScheduled) == currentNumberScheduled &&
+diff --git a/pkg/controller/daemonset/daemonset_event_handler.go b/pkg/controller/daemonset/daemonset_event_handler.go
+index 26a33f3d..68c5ec35 100644
+--- a/pkg/controller/daemonset/daemonset_event_handler.go
++++ b/pkg/controller/daemonset/daemonset_event_handler.go
+@@ -284,6 +284,23 @@ func (e *nodeEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitin
+ }
+ 
+ func (e *nodeEventHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
++	dsList := &appsv1alpha1.DaemonSetList{}
++	err := e.reader.List(context.TODO(), dsList)
++	if err != nil {
++		klog.V(4).Infof("Error enqueueing daemon sets: %v", err)
++		return
++	}
++
++	node := evt.Object.(*v1.Node)
++	for i := range dsList.Items {
++		ds := &dsList.Items[i]
++		if shouldSchedule, _ := nodeShouldRunDaemonPod(node, ds); shouldSchedule {
++			q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
++				Name:      ds.GetName(),
++				Namespace: ds.GetNamespace(),
++			}})
++		}
++	}
+ }
+ 
+ func (e *nodeEventHandler) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+diff --git a/pkg/controller/daemonset/daemonset_util.go b/pkg/controller/daemonset/daemonset_util.go
+index 420188f0..91911fbe 100644
+--- a/pkg/controller/daemonset/daemonset_util.go
++++ b/pkg/controller/daemonset/daemonset_util.go
+@@ -98,8 +98,16 @@ func nodeInSameCondition(old []corev1.NodeCondition, cur []corev1.NodeCondition)
+ //   - shouldContinueRunning:
+ //     Returns true when a daemonset should continue running on a node if a daemonset pod is already
+ //     running on that node.
+-func nodeShouldRunDaemonPod(node *corev1.Node, ds *appsv1alpha1.DaemonSet) (bool, bool) {
+-	pod := NewPod(ds, node.Name)
++func nodeShouldRunDaemonPod(node *corev1.Node, ds *appsv1alpha1.DaemonSet, countReplicasFlag ...bool) (bool, bool) {
++	var pod *corev1.Pod
++	// add standard demonset tolerations if countReplicasFlag isn't empty
++	if len(countReplicasFlag) > 0 {
++		pod = &corev1.Pod{Spec: ds.Spec.Template.Spec, ObjectMeta: ds.Spec.Template.ObjectMeta}
++		pod.Namespace = ds.Namespace
++		util.AddOrUpdateDaemonPodTolerations(&pod.Spec)
++	} else {
++		pod = NewPod(ds, node.Name)
++	}
+ 
+ 	// If the daemon set specifies a node name, check that it matches with node.Name.
```

Comment:
nodeShouldRunDaemonPod has been slightly updated so that not only is it used for `baking` a pod, but also for calculating if a daemonset's pod fits the node